lib/trino-hdfs/src/main/java/io/trino/hdfs/gcs/GcsExclusiveOutputStream.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.hdfs.gcs;
+
+import com.google.cloud.hadoop.repackaged.gcs.com.google.api.client.http.ByteArrayContent;
+import com.google.cloud.hadoop.repackaged.gcs.com.google.api.services.storage.Storage;
+import com.google.cloud.hadoop.repackaged.gcs.com.google.api.services.storage.model.StorageObject;
+import com.google.cloud.hadoop.repackaged.gcs.com.google.cloud.hadoop.gcsio.StorageResourceId;
+import io.trino.hdfs.HdfsContext;
+import io.trino.hdfs.HdfsEnvironment;
+import org.apache.hadoop.fs.Path;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+
+public class GcsExclusiveOutputStream
+        extends ByteArrayOutputStream
+{
+    private final Storage storage;
+    private final Path path;
+    private boolean closed;
+
+    public GcsExclusiveOutputStream(HdfsEnvironment environment, HdfsContext context, Path path)
```

Comment:
I was doing an experiment today with a CTAS on a Delta Lake table which resulted in a transaction log file having 1GB in size. 
Edge case obviously.
But I'm mentioning this in the context of you using `ByteArrayOutputStream`.
It goes anyway in line with the existing `TransactionLogSynchronizer`.

Reply:
I assume that the size would be fairly small since `TransactionLogSynchronizer` takes a `byte[]`. The data is already in memory in a byte array, so it shouldn't be a problem. This implementation is a hack for writing the transaction log, not a generic output stream.

Reply:
https://github.com/GoogleCloudDataproc/hadoop-connectors/blob/d570fa7a548a721bae8452c3adb6a5de6e39c714/gcsio/src/main/java/com/google/cloud/hadoop/gcsio/CreateObjectOptions.java#L37